src/deferred.js

```diff
@@ -301,7 +301,7 @@ jQuery.extend( {
 		}
 
 		// All done!
-		return deferred;
+		return this === jQuery ? deferred : jQuery.extend( this, deferred );
```

Comment:
Testing like `this instanceof jQuery.Deferred` would be more in line with our code style ([example](https://github.com/jquery/jquery/blob/9b086888b803db872e8906cbdbf0de458231f225/src/event.js#L547)), though honestly I'd be happier if all our auto-constructors were named rather than anonymous (e.g., `Deferred: function Deferred( func ) { … }`).